app/uk/gov/hmrc/entrydeclarationintervention/repositories/InterventionRepo.scala

```diff
@@ -108,60 +107,85 @@ class InterventionRepoImpl @Inject()(appConfig: AppConfig)(implicit mongo: React
-          .collect(maxDocs = -1, FailOnError[Seq[NotificationId]]()))
-      .map(_.map(_.value))
+          .find[BsonValue](equal("submissionId", submissionId))
+          .projection(fields(include("notificationId"), excludeId()))
+          .sort(ascending("receivedDateTime"))
+          .collect
+          .toFutureOption
+          .map{
+            case None => Nil
+            case Some(l) => l.map(Codecs.fromBson[NotificationId](_).value).toList
+          }
+      )
 
   def lookupIntervention(eori: String, notificationId: String): Future[Option[InterventionModel]] =
     Mdc
       .preservingMdc(
         collection
-          .find(
-            Json.obj("eori" -> eori, "notificationId" -> notificationId, "acknowledged" -> false),
-            Option.empty[JsObject])
-          .one[InterventionPersisted])
-      .map(_.map(_.toIntervention))
+          .find(and(equal("eori", eori), equal("notificationId", notificationId), equal("acknowledged", false)))
+          .first()
+          .toFutureOption()
+      ).map(_.map(_.toIntervention))
+
 
   def lookupFullIntervention(eori: String, notificationId: String): Future[Option[InterventionModel]] =
     Mdc
       .preservingMdc(
         collection
-          .find(Json.obj("eori" -> eori, "notificationId" -> notificationId), Option.empty[JsObject])
-          .one[InterventionPersisted])
-      .map(_.map(_.toIntervention))
+          .find(and(equal("eori", eori), equal("notificationId", notificationId)))
+          .first()
+          toFutureOption()
+      ).map(_.map(_.toIntervention))
 
   def acknowledgeIntervention(eori: String, notificationId: String): Future[Option[InterventionModel]] =
     Mdc
       .preservingMdc(
-        findAndUpdate(
-          query          = Json.obj("eori" -> eori, "notificationId" -> notificationId, "acknowledged" -> false),
-          update         = Json.obj("$set" -> Json.obj("acknowledged" -> true)),
-          fetchNewObject = true
-        ))
-      .map(result => result.result[InterventionPersisted].map(_.toIntervention))
+        collection.findOneAndUpdate(
+          and(equal("eori", eori), equal("notificationId", notificationId), equal("acknowledged", false)),
+          set("acknowledged", true),
+          FindOneAndUpdateOptions().returnDocument(ReturnDocument.AFTER)
+        ).toFutureOption()
+      ).map(_.map(_.toIntervention))
 
   def listInterventions(eori: String): Future[List[InterventionIds]] =
     Mdc
       .preservingMdc(
         collection
-          .find(
-            Json.obj("eori" -> eori, "acknowledged" -> false),
-            Some(Json.obj("correlationId" -> 1, "notificationId" -> 1)))
-          .sort(Json.obj("receivedDateTime" -> 1))
-          .cursor[InterventionIds]()
-          .collect[List](maxDocs = appConfig.listInterventionsLimit, err = Cursor.FailOnError[List[InterventionIds]]()))
+          .find[BsonValue](and(equal("eori", eori), equal("acknowledged", false)))
```

Comment:
If you do this as a .find[InterventionsIds) and an excludeId() to the projection, it will generate a list of InterventionIds directly. 
Will also need to add an extraCodecs repo param for InterventionIds i.e. 
extraCodecs = Seq(Codecs.playFormatCodec(InterventionIds.formats))
and In the InterventionIds companion object replace the Reads Fm with Format